src/coreclr/jit/codegenarm64.cpp

```diff
@@ -1747,66 +1654,47 @@ void CodeGen::genCaptureFuncletPrologEpilogInfo()
     // The frame size and offsets must be finalized
     assert(compiler->lvaDoneFrameLayout == Compiler::FINAL_FRAME_LAYOUT);
 
-    unsigned const PSPSize = (compiler->lvaPSPSym != BAD_VAR_NUM) ? REGSIZE_BYTES : 0;
-
-    // Because a method and funclets must have the same caller-relative PSPSym offset,
-    // if there is a PSPSym, we have to pad the funclet frame size for OSR.
-    //
-    unsigned osrPad = 0;
-    if (compiler->opts.IsOSR() && (PSPSize > 0))
-    {
-        osrPad = compiler->info.compPatchpointInfo->TotalFrameSize();
-
-        // OSR pad must be already aligned to stack size.
-        assert((osrPad % STACK_ALIGN) == 0);
-    }
-
-    genFuncletInfo.fiFunction_CallerSP_to_FP_delta = genCallerSPtoFPdelta() - osrPad;
-
     regMaskTP rsMaskSaveRegs = regSet.rsMaskCalleeSaved;
     assert((rsMaskSaveRegs & RBM_LR) != 0);
     assert((rsMaskSaveRegs & RBM_FP) != 0);
```

Comment:
Is there still a dependency on the funclet prolog to be the same as main method prolog with PSP removed?

Can this be changed to just `rsMaskSaveRegs  = RBM_LR | RBM_FP;` to avoid saving all registers that main method saves?

Reply:
My gut says that there's no dependency. There are few more places that are likely possible to optimize (https://github.com/dotnet/runtime/pull/114630#discussion_r2051683987). I'm torn between trying to address it here or separately. Even if the codegen is not perfect it was already somewhat tested by NativeAOT ABI.